Sources/Purchasing/StoreKit2/StoreKit2TransactionFetcher.swift

```diff
@@ -90,15 +111,87 @@ extension StoreKit.VerificationResult where SignedType == StoreKit.Transaction {
     var verifiedTransaction: StoreKit.Transaction? {
         switch self {
         case let .verified(transaction): return transaction
-        case .unverified: return nil
+        case let .unverified(transaction, error):
+            Logger.warn(Strings.storeKit.sk2_unverified_transaction(String(transaction.id), error))
+            return nil
         }
     }
 
     fileprivate var verifiedStoreTransaction: StoreTransaction? {
         switch self {
         case let .verified(transaction): return StoreTransaction(sk2Transaction: transaction,
                                                                  jwsRepresentation: self.jwsRepresentation)
-        case .unverified: return nil
+        case let .unverified(transaction, error):
+            Logger.warn(Strings.storeKit.sk2_unverified_transaction(String(transaction.id), error))
+            return nil
+        }
+    }
+
+}
+
+// MARK: - Private
+
+extension StoreKit2TransactionFetcher {
+
+    @available(iOS 15.0, tvOS 15.0, macOS 12.0, watchOS 8.0, *)
+    private var verifiedTransactionsJWS: [String] {
+        get async {
+            return await StoreKit.Transaction.all
+                .compactMap { $0.verifiedStoreTransaction?.jwsRepresentation }
+                .extractValues()
+        }
+    }
+
+    @available(iOS 15.0, tvOS 15.0, macOS 12.0, watchOS 8.0, *)
+    private var subscriptionStatus: [String: [Product.SubscriptionInfo.Status]] {
+        get async {
+            #if swift(>=5.9)
+            if #available(iOS 17.0, macOS 14.0, tvOS 17.0, watchOS 10.0, *) {
+                return await StoreKit.Product.SubscriptionInfo.Status.all
+                    .extractValues()
+                    .reduce(into: [:]) {result, value in
+                        result[value.groupID] = value.statuses
+                    }
+            } else {
+                // `StoreKit.Product.SubscriptionInfo.Status.all` is only available starting in iOS 17.0
```

Comment:
I'm confused, though - do you mean StoreKit.Transaction.all? Also, StoreKit.Transaction.all is available on 15+

Reply:
Check the previous branch of the if/else statement. `StoreKit.Product.SubscriptionInfo.Status.all` is new in iOS 17